tools/fuzzer/main.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ast.h"
+#include "expr_gen.h"
+
+#include "src/api.h"
+
+#include "lldb/API/SBBreakpoint.h"
+#include "lldb/API/SBDebugger.h"
+#include "lldb/API/SBError.h"
+#include "lldb/API/SBExecutionContext.h"
+#include "lldb/API/SBProcess.h"
+#include "lldb/API/SBTarget.h"
+#include "lldb/API/SBThread.h"
+#include "lldb/API/SBValue.h"
+
+#include "tools/cpp/runfiles/runfiles.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdio>
+#include <random>
+#include <string>
+#include <utility>
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+constexpr char VAR[] = "x";
+
+constexpr char LLDB_SERVER_KEY[] = "llvm_project_local/bin/lldb-server";
+constexpr char EXECUTABLE_PATH_KEY[] = "lldb_eval/testdata/fuzzer_binary";
+
+int main(int argc, char** argv) {
+  // No need to use argc when running via Bazel
+  (void) argc;
+
+  std::string error;
+  std::unique_ptr<Runfiles> runfiles(Runfiles::Create(argv[0], &error));
+  if (runfiles == nullptr) {
+    fprintf(stderr, "Could not launch the fuzzer: %s\n", error.c_str());
+    return 1;
+  }
+
+#ifndef _WIN32
+  std::string lldb_server = runfiles->Rlocation(LLDB_SERVER_KEY);
+  setenv("LLDB_DEBUGSERVER_PATH", lldb_server.c_str(), 0);
+#endif // !_WIN32
+
+  std::string exe_path = runfiles->Rlocation(EXECUTABLE_PATH_KEY);
+  std::string dirname_buf = exe_path;
+
+  std::random_device rd;
+  auto seed = rd();
+  printf("Seed for this run is: %u\n", seed);
+
+  const char* ARGV[] = { exe_path.c_str(), nullptr };
+
+  lldb::SBDebugger::Initialize();
+  {
```

Comment:
My intention is to have the destructors of all `lldb::SB*` objects run before calling `lldb::SBDebugger::Destroy()` (otherwise they will run at the end of `main()`).